blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,364 @@
+for description and example OpenSSL configuration for Platform Owner certificate
+chain, if you haven't prepared whole chain during the build process.
+
+Now switch to target platform. To make further steps easier, start by adding IP
+address to `systemd-networkd` configuration. Create two files:
+
+- `/etc/systemd/network/10-fobnail.link` (see [known issues](#known-issues)):
+
+  ```
+  [Match]
+  OriginalName=usb0
+
+  [Link]
+  Name=fobnail
+  ```
+
+- `/etc/systemd/network/10-fobnail.network`:
+
+  ```
+  [Match]
+  Name=fobnail
+
+  [Network]
+  Address=169.254.0.8/16
+  DHCPServer=false
+  ```
+
+Restart service with `sudo systemctl reload-or-restart --now systemd-networkd`.
+Connect the Token and check if it gets its IP correctly, see [known issues](#known-issues)
+if not.
+
+With Token plugged in and with correct IP we can provision the platform. Fobnail
+Token Services are available even for Attester with provisioning functions, so
+encryption key can be written in the same invocation as platform provisioning.
+Following script automates the process of creating disk image, encryption key,
+platform provisioning and writing the encryption key to the Token. Change paths
+and names to your expectations, save it and run with `sudo`:
+
+```bash
+#!/bin/bash
+
+# Some configuration variables - change as needed
+DISK_IMG=/home/user/.secure_disk.img
+DISK_SIZE_MB=128
+MNT_DIR=/home/user/secure_storage
+ATTESTER=/usr/bin/fobnail-attester-with-provisioning
+MAPPER_DEV=c1
+TMP_KEY=/tmp/keyfile.bin
+FOBNAIL_KEY_FNAME=luks_key
+
+# Get user name, regardless of whether script is started with sudo or not
+DISK_USER=${SUDO_USER:-$USER}
+
+# Create disk image and mount is as loop device
+dd if=/dev/zero of=$DISK_IMG bs=1M count=$DISK_SIZE_MB
+LOOP_DEV=`losetup -f --show $DISK_IMG`
+
+# Create encryption key and LUKS partition
+dd bs=512 count=4 if=/dev/urandom of=$TMP_KEY
+cryptsetup luksFormat --type luks2 $LOOP_DEV $TMP_KEY
+
+# Create filesystem on the partition
+cryptsetup luksOpen -d $TMP_KEY $LOOP_DEV $MAPPER_DEV
+mke2fs -j /dev/mapper/$MAPPER_DEV
+
+# Mount partition and change the owner of root directory
+mkdir -p "$MNT_DIR"
+mount /dev/mapper/${MAPPER_DEV} "$MNT_DIR"
+chown -R $DISK_USER:$DISK_USER "$MNT_DIR"
+# Optionally fill the partition with secret stuff
+# cp top_secret_file.odt "${MNT_DIR}"
+umount "$MNT_DIR"
+
+# Close LUKS partition and underlying loop device
+cryptsetup close $MAPPER_DEV
+losetup -d $LOOP_DEV
+
+# Write key to Fobnail Token and securely erase it from host
+$ATTESTER --write-file $TMP_KEY:$FOBNAIL_KEY_FNAME && \
+    dd if=/dev/urandom of=$TMP_KEY bs=$(stat -c %s $TMP_KEY) count=1 && \
+    rm $TMP_KEY
+```
+
+Platform is now provisioned and encryption key can be read by calling
+`fobnail-attester --read-file luks_key:keyfile.bin`. For better user experience
+we can automate this as well with relatively simple scripts:
```

Comment:
@krystian-hebel On this stage, the platform is provisioned and the key can be read with one command. What does "this" refer to in this sentence? What do these scripts automate? - I currently understand that reading the encrypted key

Reply:
I've changed the wording in https://github.com/3mdeb/news-and-ideas/pull/504/commits/289e3a8aa2a8c87fe55bb018ddfccbac0e03a6f3, please check if it is clear enough now.